src/GafferSceneUI/LightTool.cpp

```diff
@@ -0,0 +1,1916 @@
+				style->renderImage( Box2f( V2f( 0 ), V2f( 1.25f ) ), infoTexture.get() );
+				glPopMatrix();
+
+				glPushMatrix();
+				glTranslate( V2f( 1.75f, 0 ) );
+				style->renderText( Style::TextType::BodyText, parameterInfo, Style::NormalState, &g_hoverTextColor );
+				glPopMatrix();
+
+				if( !warningInfo.empty() )
+				{
+					glTranslate( V2f( 0, -1.5f ) );
+
+					glPushMatrix();
+					glTranslate( V2f( 0, -0.25f ) );
+					IECoreGL::ConstTexturePtr warningTexture = ImageGadget::loadTexture( "warningSmall.png" );
+					style->renderImage( Box2f( V2f( 0 ), V2f( 1.25f ) ), warningTexture.get() );
+					glPopMatrix();
+
+					glTranslate( V2f( 1.75f, 0 ) );
+					style->renderText( Style::TextType::BodyText, warningInfo, Style::NormalState, &g_hoverTextColor );
+				}
+
+				glPopMatrix();
+				glPopAttrib();
+			}
+		}
+
+	private :
+
+		void dragBegin( const DragDropEvent &event ) override
+		{
+			const auto &[ coneInspection, coneHandleAngle, penumbraInspection, penumbraHandleAngle] = spotLightHandleAngles();
+
+			m_dragStartData.coneInspection = coneInspection;
+			m_dragStartData.originalConeHandleAngle = coneHandleAngle;
+			m_dragStartData.penumbraInspection = penumbraInspection;
+			m_dragStartData.originalPenumbraHandleAngle = penumbraHandleAngle;
+
+			m_drag = AngularDrag(
+				this,
+				V3f( 0, 0, 0 ),
+				V3f( 0, 1.f, 0 ),
+				V3f( 0, 0, -1.f ),
+				event
+			);
+
+			if( getLookThroughLight() )
+			{
+				const float dragStartAngle = m_handleType == HandleType::Cone ? coneHandleAngle : penumbraHandleAngle.value();
+
+				const Line3f clickLine( event.line.p0, event.line.p1 );
+				const Line3f originLine( V3f( 0 ), V3f( 0, 0, -1.f ) );
+				const Line3f handleLine(
+					V3f( 0 ),
+					V3f( 0, 0, -1.f ) * M44f().rotate( V3f( 0, degreesToRadians( dragStartAngle ), 0 ) )
+				);
+
+				const float clickRaster = rasterDragDistance( clickLine );
+				const float originRaster = rasterDragDistance( originLine );
+				const float handleRaster = rasterDragDistance( handleLine );
+
+				const float delta = handleRaster - originRaster;
+
+				m_rasterXOffset = delta - clickRaster;
+				m_rasterZPosition = abs( delta ) / tan( degreesToRadians( dragStartAngle ) );
+			}
+		}
+
+		DragStartData spotLightHandleAngles() const
+		{
+			ScenePlug::PathScope pathScope( handleScenePath()->getContext() );
+			pathScope.setPath( &handleScenePath()->names() );
+
+			const FloatData *coneAngleData = nullptr;
+
+			Inspector::ResultPtr coneInspection = m_coneAngleInspector->inspect();
+			if( coneInspection )
+			{
+				coneAngleData = runTimeCast<const FloatData>( coneInspection->value() );
+			}
+
+			const FloatData *penumbraAngleData = nullptr;
+
+			Inspector::ResultPtr penumbraInspection = m_penumbraAngleInspector ? m_penumbraAngleInspector->inspect() : nullptr;
+			if( penumbraInspection )
+			{
+				penumbraAngleData = runTimeCast<const FloatData>( penumbraInspection->value() );
+			}
+
+			assert( coneAngleData );
```

Comment:
This assert fails/a release build will crash in the following situation:

- Activate the light tool
- Select a spotLight
- Select any non-light object such as a mesh or camera while the light tool is active

```
gaffer: src/GafferSceneUI/LightTool.cpp:1324: {anonymous}::SpotLightHandle::DragStartData {anonymous}::SpotLightHandle::spotLightHandleAngles() const: Assertion `coneAngleData' failed.

__assert_fail () from /lib64/libc.so.6
(anonymous namespace)::SpotLightHandle::spotLightHandleAngles (this=0x7f50990be470) at src/GafferSceneUI/LightTool.cpp:1324
(anonymous namespace)::SpotLightHandle::visible (this=0x7f50990be470) at src/GafferSceneUI/LightTool.cpp:863
GafferSceneUI::LightTool::updateHandleInspections (this=0x7f5093d8e960) at src/GafferSceneUI/LightTool.cpp:1689
GafferSceneUI::LightTool::preRender (this=0x7f5093d8e960) at src/GafferSceneUI/LightTool.cpp:1837
```

Reply:
Fixed in https://github.com/GafferHQ/gaffer/pull/5218/commits/61072a1f2f11066af3c9126d894dcc6f3bddf7e8